### PR TITLE
src: adds validation page

### PR DESCRIPTION
**Pull request by chris-ramon**
#### Overview
- Closes: https://github.com/bespoken/dashboard/issues/408
- [x] Adds `/skills/:sourceId/validation` page.
- [x] UI scripts page validation script form.
- [x] Wire scripts page validation script form.
- [x] UI for validation script results.

output

#### Test plan
- [x] Unit test & manual test.

### CODE REVIEW

File: src/frames/Dashboard.tsx [hunk 1]
```diff
@@ -184,9 +188,12 @@ class Dashboard extends React.Component<DashboardProps, DashboardState> {
       this.props.goTo("/skills/" + this.props.currentSource.id + "/logs");
     } else if (button.name === "integration") {
       this.props.goTo("/skills/" + this.props.currentSource.id + "/integration");
+    } else if (button.name === "scripts") {
```
Comment (jkelvie) [inline src/frames/Dashboard.tsx]: Please label this validation

File: src/index.tsx [hunk 2]
```diff
@@ -159,6 +160,7 @@ let render = function () {
                         <IndexRoute component={SourcePage} />
                         <Route path="/skills/:sourceId/logs" component={LogsPage} />
                         <Route path="/skills/:sourceId/integration" component={IntegrationPage} />
+                        <Route path="/skills/:sourceId/scripts" component={ScriptsPage} />
```
Comment (jkelvie) [inline src/index.tsx]: /validation

**Comment by chris-ramon** [comment]
@jkelvie, in regards the validation help text, I have one simple suggestion, do you have something else in mind?

image

**Comment by jkelvie** [comment]
My suggestion for the help text:

> The Validation Script has the following structure:
> `"Input": "Expected Output"`
>
> **Input**: What the user is meant to say to Alexa
> **Expected Output**: The expected response from Alexa
>
> For the expected output: 
> * If the result is audio, the expected output is compared to the transcript of the audio.
> * If the result is a stream URL, the expected output is compared to the URL.
>
> In either case, the comparison works as a "contains" - if the value of the expected output appears anywhere in the result, it is accepted.
>
> For example, in the case of a test like this:
> `"hello": "hi"`
> This test will succeed if the result is "hi", "hi there", or "oh hi"
>
> Tests are run in a series, and can build off of each other. So a sequence may be:
> `"open my skill": "welcome"`
> `"about": "I am a skill"`
> `"more": "I can do things"`
> These commands will be run one after another, as if a user was interacting with Alexa through a series of prompts.

**Comment by chris-ramon** [comment]
> My suggestion for the help text:

Def. agree on that, addressed on: https://github.com/bespoken/dashboard/pull/411/commits/42e5935f5b5e7f44c623717910f9cf3ef82e8529

image

**Comment by jkelvie** [comment]
Getting close, but a few changes:
* The link to get a token should not open in a new tab, as the users are now redirected back to the validation page
* The link should span the whole sentence (Get it here). There should be a period at the end that is *not* linked.
* Use this icon for the validation page:
https://material.io/icons/#ic_assignment_turned_in
* The validation tab should be labeled "validation (beta)" when hovering

And is this using the virtual-device.bespoken.io, or virtual-device-dev? It should be pointing at virtual-device.bespoken.io.

**Comment by jkelvie** [comment]
Additionally, are you saying quit between sequences? I think we should add one whenever we start as well.

**Comment by jkelvie** [comment]
Additionally, the state of the enable monitoring button also does not save - is this implemented?

**Comment by chris-ramon** [comment]
> Additionally, the state of the enable monitoring button also does not save - is this implemented?
not yet, should we implement it along this PR ?

**Comment by jkelvie** [comment]
I would leave it, but then hide the checkbox for now.

**Comment by chris-ramon** [comment]
> Additionally, are you saying quit between sequences? I think we should add one whenever we start as well.

between sequences we are doing:
```
await this.silentEcho.message("Alexa, pause");
```

**Comment by chris-ramon** [comment]
> And is this using the virtual-device.bespoken.io, or virtual-device-dev? It should be pointing at virtual-device.bespoken.io.

we are using [`https://virtual-device.bespoken.io/`](https://github.com/bespoken/dashboard/pull/411/files#diff-8a1c41e703a21f187f64e8c5a20980c0R135)

**Comment by chris-ramon** [comment]
> Getting close, but a few changes:
> 
> The link to get a token should not open in a new tab, as the users are now redirected back to the validation page
> The link should span the whole sentence (Get it here). There should be a period at the end that is not  linked.
> Use this icon for the validation page:
> https://material.io/icons/#ic_assignment_turned_in
> The validation tab should be labeled "validation (beta)" when hovering

addressed via: https://github.com/bespoken/dashboard/pull/411/commits/7840fc9dfe3fd359da2a9dc49d082de8811e8871

**Comment by chris-ramon** [comment]
screen shot 2017-09-08 at 11 52 27 am

addressed via: https://github.com/bespoken/dashboard/pull/411/commits/6b5d51f3d5797df18c6311179aee47c3b79becee

**Comment by chris-ramon** [comment]
very long validation results, overlaps container, now addressed on: https://github.com/bespoken/dashboard/pull/411/commits/77f3e3c6132fd2f1343aef7d8090e7edbf479bb9
image

**Comment by chris-ramon** [comment]
@jkelvie comments addressed on latest changes, looking forward to :ship: validation :smile: 

> Getting close, but a few changes:
> 
> The link to get a token should not open in a new tab, as the users are now redirected back to the validation page

output

***

> The link should span the whole sentence (Get it here). There should be a period at the end that is not  linked.

image

***

> Use this icon for the validation page:
> https://material.io/icons/#ic_assignment_turned_in
> The validation tab should be labeled "validation (beta)" when hovering

  image

***

> And is this using the virtual-device.bespoken.io, or virtual-device-dev? It should be pointing at virtual-device.bespoken.io.

We are using [virtual-device](https://github.com/bespoken/dashboard/pull/411/files#diff-8a1c41e703a21f187f64e8c5a20980c0R136).

**Comment by jkelvie** [comment]
Okay, these changes looking good. Two more items:
1) Hide the enable monitoring button
2) Preserve the last script the user entered (this can be done simply by cookie)

And does this version include the latest changes to the silent-echo-sdk?

**Comment by chris-ramon** [comment]
> Okay, these changes looking good. Two more items:
> 
> Hide the enable monitoring button

image

***

> Preserve the last script the user entered (this can be done simply by cookie)

Neat feature, but I think we should go for locaStorage, since the data is only needed client-side by now, cookie would be better go-to if data is needed server-side as well, implemented using localStorage:

output

***

> And does this version include the latest changes to the silent-echo-sdk?

Good call!, just released new version of `silent-echo-sdk` and updated it on `source-api`: https://github.com/bespoken/source-api/pull/22

**Comment by chris-ramon** [comment]
> When you first come the page, it shows null in the script box for me:
screen shot 2017-09-12 at 3 24 15 pm

> When it is empty, can we show an example (like the we study billionaries one) that goes away when you click inside?

image

**Comment by chris-ramon** [comment]
Saving last script per source, addressed on: https://github.com/bespoken/dashboard/pull/411/commits/0be1c01324c49c421e183ef309d633163eddbdd9
cc/ @jkelvie 

image

**Pull request merged**